### ISSUE TITLE
Spector is hiding a bug somehow

### ISSUE THREAD

**Issue by Diniden** [BabylonJS/Spector.js#186]
Spector has been wonderful for finding my problems. So I resolved all of my issues with FLOAT textures and binding them as a render target to an FBO. Works great and I get the correct rendering on my screen.

So, I start to benchmark performance, thus I disable Spector for max benefit.

Lo! My FLOAT textures don't work as my FBOs are now saying they are invalid (with no GL_ERROR output at all).

This is what spector says my commands are running:

```
createTexture -> WebGLTexture - ID: 0
bindTexture: TEXTURE_2D, WebGLTexture - ID: 0
texParameteri: TEXTURE_2D, TEXTURE_MAG_FILTER, NEAREST
texParameteri: TEXTURE_2D, TEXTURE_MIN_FILTER, NEAREST
texParameteri: TEXTURE_2D, TEXTURE_WRAP_S, CLAMP_TO_EDGE
texParameteri: TEXTURE_2D, TEXTURE_WRAP_T, CLAMP_TO_EDGE
pixelStorei: UNPACK_PREMULTIPLY_ALPHA_WEBGL, false
pixelStorei: UNPACK_FLIP_Y_WEBGL, false
texImage2D: TEXTURE_2D, 0, RGBA, 2232, 1896, 0, RGBA, UNSIGNED_BYTE, null
activeTexture: TEXTURE1
createTexture -> WebGLTexture - ID: 1
bindTexture: TEXTURE_2D, WebGLTexture - ID: 1
texParameteri: TEXTURE_2D, TEXTURE_MAG_FILTER, NEAREST
texParameteri: TEXTURE_2D, TEXTURE_MIN_FILTER, NEAREST
texParameteri: TEXTURE_2D, TEXTURE_WRAP_S, CLAMP_TO_EDGE
texParameteri: TEXTURE_2D, TEXTURE_WRAP_T, CLAMP_TO_EDGE
pixelStorei: UNPACK_PREMULTIPLY_ALPHA_WEBGL, false
pixelStorei: UNPACK_FLIP_Y_WEBGL, false
texImage2D: TEXTURE_2D, 0, RGBA, 2232, 1896, 0, RGBA, UNSIGNED_BYTE, null
activeTexture: TEXTURE2
createTexture -> WebGLTexture - ID: 2
bindTexture: TEXTURE_2D, WebGLTexture - ID: 2
texParameteri: TEXTURE_2D, TEXTURE_MAG_FILTER, NEAREST
texParameteri: TEXTURE_2D, TEXTURE_MIN_FILTER, NEAREST
texImage2D: TEXTURE_2D, 0, RGBA32F, 2232, 1896, 0, RGBA, FLOAT, null
activeTexture: TEXTURE3
createTexture -> WebGLTexture - ID: 3
bindTexture: TEXTURE_2D, WebGLTexture - ID: 3
texParameteri: TEXTURE_2D, TEXTURE_MAG_FILTER, NEAREST
texParameteri: TEXTURE_2D, TEXTURE_MIN_FILTER, NEAREST
texParameteri: TEXTURE_2D, TEXTURE_WRAP_S, CLAMP_TO_EDGE
texParameteri: TEXTURE_2D, TEXTURE_WRAP_T, CLAMP_TO_EDGE
pixelStorei: UNPACK_PREMULTIPLY_ALPHA_WEBGL, false
pixelStorei: UNPACK_FLIP_Y_WEBGL, false
texImage2D: TEXTURE_2D, 0, DEPTH_COMPONENT16, 2232, 1896, 0, DEPTH_COMPONENT, UNSIGNED_SHORT, null
createFramebuffer -> WebGLFramebuffer - ID: 0
bindFramebuffer: FRAMEBUFFER, WebGLFramebuffer - ID: 0
framebufferTexture2D: FRAMEBUFFER, COLOR_ATTACHMENT0_WEBGL, TEXTURE_2D, WebGLTexture - ID: 0, 0
framebufferTexture2D: FRAMEBUFFER, COLOR_ATTACHMENT1_WEBGL, TEXTURE_2D, WebGLTexture - ID: 1, 0
framebufferTexture2D: FRAMEBUFFER, COLOR_ATTACHMENT2_WEBGL, TEXTURE_2D, WebGLTexture - ID: 2, 0
framebufferTexture2D: FRAMEBUFFER, DEPTH_ATTACHMENT, TEXTURE_2D, WebGLTexture - ID: 3, 0
checkFramebufferStatus: FRAMEBUFFER -> FRAMEBUFFER_COMPLETE
```

Works perfect when spector is enabled.

Disable spector and I get:

```
FRAMEBUFFER_INCOMPLETE_ATTACHMENT
gl-proxy.ts:1207 GL Error: No Error
```

So, spector is somehow masking my commands and fixing something ever so slightly?
Any insights into this? Or point me toward anything that spector is "massaging" before submitting as GL calls?

And just to assert: I have narrowed it down to specifically be my FLOAT texture I am trying to attach to the FBO. Everything works fine without it in the pipeline.

BROWSERS: Firefox, Chrome
OS: OSX
Dedicated GPU

Oddly, it works fine in safari...

**Comment by sebavan** [comment]
I am wondering if you use webgl1 and forgot to enable some extensions for float support where spector would request them on your behalf resulting in not color renderable texture hence the completness issue ?

**Comment by Diniden** [comment]
@sebavan It is definitely not running webgl1. There are a few things that would indicate it's use of WebGL 1 (Such as picking RGBA instead of RGBA32F) and if RGBA was used the error shows a bit more correctly.

Also, I have extensive feedback from my system showing WebGL2 is in use and any necessary extensions are automatically called and enabled at start up.

**Comment by sebavan** [comment]
Ok, this would have been to easy :-) Do you have a repro I could look into ?

**Comment by Diniden** [comment]
I am working on a "minimalistic" repro for you. If you want something quick and easy to play with:

clone git@github.com:vega-studio/deltav.git
go to the dev branch
then `npm run dev`
then localhost:8080 -> lookit the "deferred shading" demo available.

If you don't want to bother with that...we'll see how my minimalist repro goes....

**Comment by sebavan** [comment]
perfect, I ll wait on the minimalistic one and look into it as soon as I ll receive it.

**Comment by Diniden** [comment]
@sebavan Took me a bit to get spector to recognize the context (still unsure why), but here's a working example that shows spector magically fixes the Float texture. I put this in index.html and had to host the file from a server to get Spector to enable. But after that, the issue shows up nicely.


[index.txt](https://github.com/BabylonJS/Spector.js/files/6045747/index.txt)

**Comment by sebavan** [comment]
Thanks !!!! I ll have a look tonight

**Comment by Diniden** [comment]
@sebavan Nevermind -facepalm-

Don't need to waste your time. I found the issue: WebGL2 also has a required extension for rendering to a FLOAT texture WEBGL_color_buffer_float. I must have missed this extension before : /

That being said: I don't see the benefit of having Spector call extensions itself. It definitely masked an issue I had without spector. I don't know all the reasons behind your guys reasoning though :) just a thought

Thanks for the rubber ducky debugging -more facepalm-

**Comment by sebavan** [comment]
Yup I longly struggled with this approach but the only way to no if it is supported is to query it :-)

I should maybe query it on another canvas, and report if enabled or not.

I ll think about it for the next update, thanks
